### PR TITLE
[Arc] No need to double render server-side.

### PR DESCRIPTION
**Pull request by alloy**
Closes https://artsyproduct.atlassian.net/browse/PLATFORM-1838

found-relay will fetch the queries for the current route
during `getFarceResult`, which is kicked-off before
ever rendering the component tree. This means that
only nested QueryRenderers would not get kicked off,
which isn’t something we want anyways.

----

## Force before

Median benchmarking of Artwork page: 2798ms

double-render-pass

## Force after

Median benchmarking of Artwork page: 1521ms

single-render-pass

### CODE REVIEW

File: packages/arc/src/buildServerApp.tsx [hunk 1]
```diff
@@ -73,26 +73,21 @@ export function buildServerApp(config: RouterConfig): Promise<Resolve> {
...
+        sheet.collectStyles(<ServerApp />)
+      )
 
       // Get cache from middleware
       let relayData = {}
       if (relayEnvironment.relaySSRMiddleware) {
         relayData = await relayEnvironment.relaySSRMiddleware.getCache()
       }
 
-      // Clean relay response and remoove unneeded data structures
```
Comment (zephraph) [inline packages/arc/src/buildServerApp.tsx]: 🐮

Comment (alloy) [reply packages/arc/src/buildServerApp.tsx]:
```
 _   _
((___))
[ x x ]
 \   /
 (' ')
  (U)
```

Comment (alloy) [reply packages/arc/src/buildServerApp.tsx]: (☝️ That’s from Cult of the Dead Cow btw.)

Comment (zephraph) [reply packages/arc/src/buildServerApp.tsx]: Wasn't that an old school warez/hacking collective?

**Comment by alloy** [comment]
@zephraph That’s correct 👍

**Comment by alloy** [comment]
Cool, going to self-merge then so I can be around during testing.

**Comment by artsyit** [comment]
:rocket: PR was released in v1.4.5 :rocket:

**Comment by joeyAghion** [comment]
👀 This seems potentially huge, but I'm not sure I understand. Does this address the server-side double-rendering we've discussed a few times, or is it a separate optimization?

**Comment by alloy** [comment]
> Does this address the server-side double-rendering we've discussed a few times, or is it a separate optimization?

It might sometimes have been conflated with other things, but yes–when @damassi and I talk about double-rendering we’re referring to this.

File: packages/arc/src/buildServerApp.tsx (hunk 1 above)
Comment (damassi) [reply packages/arc/src/buildServerApp.tsx]: haha Beto :)

**Comment by damassi** [comment]
So @alloy - was this never needed?

**Comment by alloy** [comment]
@damassi I suspect this was needed when there were waterfall requests being made either for real or interpreted as such? But otherwise purely for route queries this shouldn’t have been needed afaik.

**Comment by damassi** [comment]
👍

File: packages/arc/src/buildServerApp.tsx (hunk 1 above)
Comment (alloy) [reply packages/arc/src/buildServerApp.tsx]: lol, yeah man, they’re everywhere

**Pull request merged**